xbmc/addons/kodi-dev-kit/include/kodi/c-api/addon-instance/imagedecoder.h

```diff
@@ -0,0 +1,451 @@
+/*
+ *  Copyright (C) 2005-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#ifndef C_API_ADDONINSTANCE_IMAGE_DECODER_H
+#define C_API_ADDONINSTANCE_IMAGE_DECODER_H
+
+#include "../addon_base.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif /* __cplusplus */
+
+  typedef void* KODI_ADDON_IMAGEDECODER_HDL;
+
+  //============================================================================
+  /// @defgroup cpp_kodi_addon_imagedecoder_Defs_ADDON_IMG_FMT enum ADDON_IMG_FMT
+  /// @ingroup cpp_kodi_addon_imagedecoder_Defs
+  /// @brief **Image format types**\n
+  /// Used to define wanted target format where image decoder should give to
+  /// Kodi.
+  ///
+  ///@{
+  typedef enum ADDON_IMG_FMT
```

Comment:
Are there other image formats possible? We define pixel layouts quite often, is it possible to use or expand an existing pixel layout enum?

Reply:
Thinking about it, it's probably better to separate image pixel formats from other pixel formats, as images have different requirements than video of games for example. So disregard.

Reply:
> Are there other image formats possible? We define pixel layouts quite often, is it possible to use or expand an existing pixel layout enum?

Not really the enum is only in relationship to Kodi's one here: https://github.com/xbmc/xbmc/blob/master/xbmc/guilib/TextureFormats.h#L18_L21, any others are currently not there (them below there a fixed to Windows and unusable by addon).

Them on game by `GAME_PIXEL_FORMAT` relates to VideoPlayers `AVPixelFormat`.

> Thinking about it, it's probably better to separate image pixel formats from other pixel formats, as images have different requirements than video of games for example. So disregard.

:+1: